lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -0,0 +1,167 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import asyncio
+import datetime
+import logging
+
+import grpc
+from lte.protos.s6a_service_pb2 import DeleteSubscriberRequest
+from lte.protos.s6a_service_pb2_grpc import S6aServiceStub
+from lte.protos.subscriberdb_pb2 import (ListSubscribersRequest,
+                                         LTESubscription, SubscriberData)
+from magma.common.grpc_client_manager import GRPCClientManager
+from magma.common.rpc_utils import grpc_async_wrapper
+from magma.common.sdwatchdog import SDWatchdogTask
+from magma.common.service_registry import ServiceRegistry
+from magma.subscriberdb.store.sqlite import SqliteStore
+from magma.subscriberdb.metrics import SUBSCRIBER_SYNC_LATENCY, \
+    SUBSCRIBER_SYNC_SUCCESS_TOTAL, SUBSCRIBER_SYNC_FAILURE_TOTAL
+
+
+class SubscriberDBCloudClient(SDWatchdogTask):
+    """
+    SubscriberDBCloudClient for requesting subscribers from Orchestrator.
+
+    SubscriberDBCloudClient calls the Orchestrator's SubscriberDBCloud service
+    to fetch pages of subscribers stored in Orchestrator and updates the local
+    subscriberdb store with the changes.
+    """
+    DEFAULT_SUBSCRIBERDB_SYNC_INTERVAL = 180
+    DEFAULT_SUBSCRIBER_PAGE_SIZE = 250
+    SUBSCRIBERDB_REQUEST_TIMEOUT = 10
+
+    def __init__(self,
+                 loop: asyncio.AbstractEventLoop,
+                 store: SqliteStore,
+                 subscriber_page_size: int,
+                 sync_interval: int,
+                 grpc_client_manager: GRPCClientManager):
+
+        super().__init__(
+            sync_interval,
+            loop
+        )
+        self._loop = loop
+        self._subscriber_page_size = subscriber_page_size
+        self._store = store
+
+        # grpc_client_manager to manage grpc client recycling
+        self._grpc_client_manager = grpc_client_manager
+
+    async def _run(self) -> None:
```

Comment:
Where are we enforcing the sync_interval ?

Reply:
It's enforced via SDWatchdog and initialized from `super().__init__(
            sync_interval,`